sirepo/package_data/static/js/radia.js

```diff
@@ -1030,6 +1069,19 @@ SIREPO.app.controller('RadiaSourceController', function (appState, geometry, pan
                         self.selectedObject = null;
                     }
                 }
+                if (o.type === 'racetrack') {
+                    // calculate the size
```

Comment:
How does one create an object of this type (racetrack)? It isn't in shapeSelector.availableTypes so I'm not sure how to exercise this code.

Reply:
create a  new freehand simulation and go to the design tab. There is a racetrack icon in the toolbar you can drag and drop into the workspace:

<img width="411" alt="Screen Shot 2021-12-15 at 9 43 03 AM" src="https://user-images.githubusercontent.com/30352290/146228025-ec0240ed-022d-4ec8-9d39-18bd38086d11.png">

(there is a bug where the shape does not immediately show up when you drop it (issue forthcoming). Go back to the viz tab to see the coil)

Reply:
ah thanks!